experimental/personnames/src/provider.rs

```diff
@@ -0,0 +1,115 @@
+derive(serde::Serialize, databake::Bake),
+databake(path = icu_personnames::provider))
+]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+pub struct PersonNamesFormattingDefinitionV1<'data> {
+    /// <nameOrderLocales order="surnameFirst">ko vi yue zh</nameOrderLocales>
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub surname_first_locales: VarZeroVec<'data, str>,
+
+    /// <nameOrderLocales order="givenFirst">und en</nameOrderLocales>
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub given_first_locales: VarZeroVec<'data, str>,
+
+    /// foreignSpaceReplacement element.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub foreign_space_replacement: Option<Cow<'data, str>>,
+
+    /// Equivalent of initialPattern tag + initial
+    /// ```xml
+    /// <initialPattern type="initial">{0}.</initialPattern>
+    /// ```
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub initial_pattern: Option<Cow<'data, str>>,
+
+    /// Equivalent of initialPattern tag + initialSequence
+    /// ```xml
+    /// <initialPattern type="initialSequence">{0} {1}</initialPattern>
+    /// ```
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub initial_pattern_sequence: Option<Cow<'data, str>>,
+
+    /// Equivalent of PersonNames
+    /// ```xml
+    /// <personName>...</personName>
+    /// ```
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub person_names_patterns: VarZeroVec<'data, PersonNamesFormattingDataVARULE>,
+}
+
+/// Person Name Attributes.
+/// {order=givenFirst, length=long, usage=referring, formality=formal}
+/// see https://www.unicode.org/reports/tr35/tr35-personNames.html#personname-element
+#[repr(u32)]
+pub enum PersonNamesFormattingAttributes {
+    // Order
+    GivenFirst = 1 << 0,
+    SurnameFirst = 1 << 1,
+    Sorting = 1 << 2,
+    // Length
+    Short = 1 << 3,
+    Medium = 1 << 4,
+    Long = 1 << 5,
+    // Usage
+    Addressing = 1 << 6,
+    Referring = 1 << 7,
+    Monogram = 1 << 8,
+    // Formality
+    Formal = 1 << 9,
+    Informal = 1 << 10,
+}
+
+type PersonNamesFormattingAttributesMask = u32;
+
+/// PersonName Formatting data.
+///
+/// https://www.unicode.org/reports/tr35/tr35-personNames.html#personname-element
+#[zerovec::make_varule(PersonNamesFormattingDataVARULE)]
+#[zerovec::skip_derive(ZeroMapKV, Ord)]
+#[cfg_attr(feature = "datagen",
+derive(serde::Serialize, databake::Bake),
+databake(path = icu_personnames::provider),
+zerovec::derive(Serialize))
+]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Deserialize),
+    zerovec::derive(Deserialize)
+)]
+pub struct PersonNamesFormattingData<'data> {
+    /// Attributes
+    /// https://www.unicode.org/reports/tr35/tr35-personNames.html#personname-element
+    pub attributes: PersonNamesFormattingAttributesMask,
+    /// https://www.unicode.org/reports/tr35/tr35-personNames.html#namepattern-syntax
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub patterns: VarZeroVec<'data, str>,
```

Comment:
Note: We'll want to adopt `icu_pattern` when available, but until then, do what you want with patterns and we'll refactor later. (CC @zbraniecki for another use case)